### ISSUE TITLE
Sample temperature and magnetic_field

### ISSUE THREAD

**Issue by g-guenther** [nexusformat/definitions#1340]
Currently, in the documentation [temperature](https://manual.nexusformat.org/classes/base_classes/NXsample.html#nxsample-temperature-field) is of type field while magnetic_field exists as type [field](https://manual.nexusformat.org/classes/base_classes/NXsample.html#nxsample-magnetic-field-field) and as [group NXlog](https://manual.nexusformat.org/classes/base_classes/NXsample.html#nxsample-magnetic-field-group). According to #816, I guess:

- temperature should be of type NXlog
- magnetic_field of type field should be removed
- maybe electric_field, stress_field, pressure should become NXlog as well

**Comment by mkoennecke** [comment]
No, this is something which NeXus introduced with streaming support. Any field can either be a single number or array or a NXlog. The single number or array is applicable when there is only a single value per scan point. However if you do event streaming and the field is changing during data acquisition then it becomes a NXlog. Consider a fast tomography experiment where you move the sample rotation while collecting images and trying to associate sample positions with images  through time later on. In this case both the sample rotation and the detector images live in NXlog's. 

Od you want to monitor temperature through a long running data acquisition.

**Comment by g-guenther** [comment]
Thanks for the explanation. If I understand you correctly, some corresponding NXlog classes are missing in the current documentation: temperature, electric_field, stress field, and pressure should be listed as NXlog, as well. Is this what you mean?

**Comment by mkoennecke** [comment]
Well, this is documented at https://manual.nexusformat.org/rules.html under the heading Streaming Data Acquisition and Logging as a general rule. IMHO, this is enough and may not need to be repeated in the base classes.
